website/content/docs/api-gateway/configuration/gatewayclassconfig.mdx

```diff
@@ -0,0 +1,56 @@
+---
+layout: docs
+page_title: Consul API Gateway GatewayClassConfig
+description:  >-
+  Consul API Gateway GatewayClassConfig
+---
+
+# GatewayClassConfig
+
+The `GatewayClassConfig` object describes Consul API Gateway-related configuration parameters for the [`GatewayClass`](#gatewayclass).
+
+Add the `kind: GatewayClassConfig` option to the gateway values file to declare a gateway class.
+The following example creates a gateway class configuration called `test-gateway-class-config`:
+
+<CodeBlockConfig filename="gateway.yaml">
+
+  ```yaml
+  apiVersion: api-gateway.consul.hashicorp.com/v1alpha1
+  kind: GatewayClassConfig
+  metadata:
+    name: test-gateway-class-config
+  spec:
+    useHostPorts: true
+    logLevel: 'trace'
+    consul:
+      scheme: 'https'
+      ports:
+        http: 8501
+        grpc: 8502
+```
+</CodeBlockConfig>
+
+The following table describes the allowed parameters for the `spec` array:
+
+| Parameter  | Description | Type | Default |
+| ---        | ---         | ---- | ------- |
+| `consul.address`                  | Specifies the address of the Consul server to communicate with in the gateway pod. If unspecified, the pod will attempt to use a local agent on the host on which the pod is running.                                                                                                                                                                                               | String  | N/A                                              |
+| `consul.authentication.account`   | Specifies the Kubernetes service account to use for authentication.                                                                                                                                                                                                                                                                                                                 | String  | N/A                                              |
+| `consul.authentication.managed`   | Set to `true` to enable deployments to run with managed service accounts created by the gateway controller. The `consul.authentication.account` field is ignored when this option is enabled.                                                                                                                                                                                       | Boolean | `false`                                          |
```

Comment:
It's part of an operator pattern people who are used to kubernetes and installing crds would have likely run into before, but it might not be a bad idea to add a section explaining what it is.